File: backend/app/schemas/project.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from enum import Enum

class ProjectStatus(str, Enum):
    DRAFT = "draft"
    UPLOADED = "uploaded"
    PARSING = "parsing"
    PARSED = "parsed"
    OPTIMIZING = "optimizing"
    READY = "ready"
    EXPORTED = "exported"
    ARCHIVED = "archived"

class ProjectVisibility(str, Enum):
    PRIVATE = "private"
    TEAM = "team"
    PUBLIC = "public"

class ProjectCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    industry: Optional[str] = None
    language: str = "ru"
    visibility: Optional[ProjectVisibility] = ProjectVisibility.PRIVATE

class ProjectUpdate(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None
    industry: Optional[str] = None
    language: Optional[str] = None
    visibility: Optional[ProjectVisibility] = None
    status: Optional[ProjectStatus] = None

class ProjectResponse(BaseModel):
    id: str
    title: str
    description: Optional[str]
    status: ProjectStatus
    template_id: Optional[str]
    industry: Optional[str]
    language: str
    visibility: ProjectVisibility
    created_at: datetime
    updated_at: datetime
    last_exported_at: Optional[datetime]
    
    class Config:
        orm_mode = True

class PaginationParams(BaseModel):
    total: int
    skip: int
    limit: int
    has_more: bool

class ProjectListResponse(BaseModel):
    data: List[ProjectResponse]
    pagination: PaginationParams

File: frontend/src/services/project.ts
import api from './api'

export interface Project {
  id: string
  title: string
  description: string | null
  status: 'draft' | 'uploaded' | 'parsing' | 'parsed' | 'optimizing' | 'ready' | 'exported' | 'archived'
  template_id: string | null
  industry: string | null
  language: string
  visibility: 'private' | 'team' | 'public'
  created_at: string
  updated_at: string
  last_exported_at: string | null
}

export interface ProjectCreateData {
  title: string
  description?: string
  industry?: string
  language?: string
  visibility?: 'private' | 'team' | 'public'
}

export interface ProjectUpdateData {
  title?: string
  description?: string
  industry?: string
  language?: string
  visibility?: 'private' | 'team' | 'public'
  status?: Project['status']
}

export const projectService = {
  // Получить список проектов
  getProjects: async (params?: {
    page?: number
    limit?: number
    status?: Project['status']
    sort?: string
  }) => {
    const response = await api.get('/api/v1/projects', { params })
    return response.data
  },

  // Создать проект
  createProject: async (data: ProjectCreateData) => {
    const response = await api.post('/api/v1/projects', data)
    return response.data
  },

  // Получить проект по ID
  getProject: async (id: string) => {
    const response = await api.get(`/api/v1/projects/${id}`)
    return response.data
  },

  // Обновить проект
  updateProject: async (id: string, data: ProjectUpdateData) => {
    const response = await api.put(`/api/v1/projects/${id}`, data)
    return response.data
  },

  // Удалить проект
  deleteProject: async (id: string) => {
    await api.delete(`/api/v1/projects/${id}`)
  },

  // Загрузить файл презентации
  uploadPresentation: async (projectId: string, file: File) => {
    const formData = new FormData()
    formData.append('file', file)
    
    const response = await api.post(`/api/v1/files/${projectId}/upload`, formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    })
    
    return response.data
  },

  // Получить версии проекта
  getProjectVersions: async (projectId: string) => {
    const response = await api.get(`/api/v1/files/${projectId}/versions`)
    return response.data
  },
}
